### PR TITLE
update load genotyping plates script

### PR DESCRIPTION
**Pull request by lukasmueller**
Description 
-----------------------------------------------------

Checklist 
---------------------------------------------------------------------------------------
- [ ] Refactoring only
- [ ] Documentation only
- [ ] Fixture update only
- [ ] Bug fix
  - [ ] The relevant issue has been closed.
  - [ ] Further work is required.
- [ ] New feature
  - [ ] Relevant tests have been created and run.
  - [ ] Data was added to the fixture
    - [ ] Data was added via a patch in `/t/data/fixture/patches/`.
  - [ ] User-Facing Change
    - [ ] The user manual in `/docs` has been updated.
  - [ ] Any new Perl has been documented using **perldoc**.
  - [ ] Any new JavaScript has been documented using **JSDoc**.
  - [ ] Any new _legacy_ JavaScript has been moved from `/js` to `/js/source/legacy`.

### CODE REVIEW

File: bin/load_genotyping_plates.pl [hunk 1]
```diff
@@ -319,29 +309,43 @@ =head2 AUTHORS
 
 my $coderef= sub  {
 
+    my $trial = CXGN::Trial->new( { bcs_schema => $schema, trial_id => $genotyping_project_id });
+    my $location_data = $trial->get_location();
+    my $location_id = $location_data->[0];
```
Comment (titima15) [inline bin/load_genotyping_plates.pl:314]: $location_id can be removed (using only $location_name)

File: bin/load_genotyping_plates.pl [hunk 2]
```diff
@@ -319,29 +309,43 @@ =head2 AUTHORS
...
+    my $location_name = $location_data->[1];
+    my $description = $trial->get_description();
+    my $genotyping_facility = $trial->get_genotyping_facility();
+    my $plate_year = $trial->get_year();
+    
+    my $program_object = CXGN::BreedersToolbox::Projects->new( { schema => $schema });
+    my $breeding_program_data = $program_object->get_breeding_programs_by_trial($genotyping_project_id);
+    my $breeding_program_id = $breeding_program_data->[0]->[0];
```
Comment (titima15) [inline bin/load_genotyping_plates.pl:322]: $breeding_program_id can be removed (using only $breeding_program_name)

**Review by titima15 — COMMENTED** [review]
Also, looks like there is no data validation step in the script?

File: bin/load_genotyping_plates.pl [hunk 3]
```diff
@@ -5,7 +5,9 @@
 
 =head1 SYNOPSIS
 
-load_genotyping_plates.pl  -H [dbhost] -D [dbname] -i inFile -b [breeding program name] -u [username] -g genotyping_project -l location [-t]
+NOTE: You need to create the genotyping project in the database first. With the -g option, provide the name of genotyping project the plates should be associated with. Metadata such as year and location will be loaded from the genotyping object directly.
+
+load_genotyping_plates.pl  -H [dbhost] -D [dbname] -i inFile -b [breeding program name] -u [username] -g genotyping_project [-t] -f format
```
[the remark this answers is not in the document]
Comment (lukasmueller) [reply bin/load_genotyping_plates.pl:10]: Removed!

File: bin/load_genotyping_plates.pl [hunk 4]
```diff
@@ -111,12 +105,11 @@ =head2 AUTHORS
...
-my ( $help, $dbhost, $dbname, $infile, $sites, $types, $test, $username, $breeding_program_name, $genotyping_project, $location, $year, $format );
+my ( $help, $dbhost, $dbname, $infile, $sites, $types, $test, $username, $breeding_program_name, $genotyping_project, $location, $format );
 GetOptions(
     'i=s'        => \$infile,
     'b=s'        => \$breeding_program_name,
     'l=s'        => \$location,
-    'y=s'        => \$year,
     'g=s'        => \$genotyping_project,
     't'          => \$test,
     'f=s'        => \$format,
```
[the remark this answers is not in the document]
Comment (lukasmueller) [reply bin/load_genotyping_plates.pl:115]: DONE!

File: bin/load_genotyping_plates.pl [hunk 5]
```diff
@@ -319,29 +309,43 @@ =head2 AUTHORS
...
-							 });
+	my $trial_create = CXGN::Trial::TrialCreate->new(
+	    {
+		chado_schema      => $schema,
+		dbh               => $dbh,
+		design_type       => 'genotyping_plate',
+		design            => $trial_design_hash{$trial_name},
+		program           => $breeding_program_name,
+		trial_year        => $plate_year,
+		trial_description => $trial_name,
```
[the remark this answers is not in the document]
Comment (lukasmueller) [reply bin/load_genotyping_plates.pl:336]: DONE!

File: bin/load_genotyping_plates.pl [hunk 6]
```diff
@@ -319,29 +309,43 @@ =head2 AUTHORS
...
+		trial_location    => $location_name,
+		trial_name        => $trial_name,
+		operator          => $operator,
+		owner_id           => $sp_person_id,
+		is_genotyping      => 1,
+		genotyping_user_id => $sp_person_id,
+		genotyping_plate_format => $format,
+		genotyping_plate_sample_type => 'accession',
```
[the remark this answers is not in the document]
Comment (lukasmueller) [reply bin/load_genotyping_plates.pl:344]: I think that might not be so easy to add right now

File: bin/load_genotyping_plates.pl (hunk 1 above)
Comment (lukasmueller) [reply bin/load_genotyping_plates.pl:314]: DONE.

File: bin/load_genotyping_plates.pl (hunk 2 above)
Comment (lukasmueller) [reply bin/load_genotyping_plates.pl:322]: done

**Review by titima15 — APPROVED** [review]
👍

**Review by MFlores2021 — APPROVED** [review]

**Pull request merged**